pymc/model.py

```diff
@@ -1507,7 +1509,7 @@ def add_named_variable(self, var, dims: Optional[Tuple[Union[str, None], ...]] =
             for dim in dims:
                 if dim not in self.coords and dim is not None:
                     raise ValueError(f"Dimension {dim} is not specified in `coords`.")
-            if any(var.name == dim for dim in dims):
+            if any(var.name == dim != None for dim in dims):
```

Comment:
```suggestion
            if any(var.name == dim for dim in dims if dim is not None):
```

Reply:
My code worked too, but OK let's have it your way ;)